# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/*/ccache/dont_stat_compiler.patch
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./ccache.c.orig	Mon Feb 17 01:11:58 2003
+++ ./ccache.c	Tue Feb 18 20:10:52 2003
@@ -290,13 +290,15 @@
 
 	/* the compiler driver size and date. This is a simple minded way
 	   to try and detect compiler upgrades. It is not 100% reliable */
-	if (stat(args->argv[0], &st) != 0) {
-		cc_log("Couldn't stat the compiler!? (argv[0]='%s')\n", args->argv[0]);
-		stats_update(STATS_COMPILER);
-		failed();
+	if (!getenv("CCACHE_DONT_STAT_COMPILER")) {
+		if (stat(args->argv[0], &st) != 0) {
+			cc_log("Couldn't stat the compiler!? (argv[0]='%s')\n", args->argv[0]);
+			stats_update(STATS_COMPILER);
+			failed();
+		}
+		hash_int(st.st_size);
+		hash_int(st.st_mtime);
 	}
-	hash_int(st.st_size);
-	hash_int(st.st_mtime);
 
 	/* possibly hash the current working directory */
 	if (getenv("CCACHE_HASHDIR")) {
